sim/battle.ts

```diff
@@ -1165,6 +1165,9 @@ export class Battle {
 				if (!side.pokemonLeft) continue;
 				const activeData = side.active.map(pokemon => pokemon?.getMoveRequestData());
 				requests[i] = {active: activeData, side: side.getRequestData()};
+				if (this.gameType === 'multi') {
+					requests[i].ally = side.allySide?.getRequestData(true);
```

Comment:
```suggestion
					requests[i].ally = side.allySide!.getRequestData(true);
```

Honestly this should definitely crash if `allySide` doesn't exist.